scripts/delete_vm.py

```diff
@@ -275,17 +272,16 @@ def vsphere_check_with_api(api, run_time, text):
         run_time: when this run time is exceeded for the VM, it will be deleted
         text: when this string is found in the name of VM, it may be deleted
     """
-    vms = VirtualMachine.all(api.api)
+    vms = api.all_vms()
     vms_to_delete = []
     templates_to_delete = []
 
-    nightly_templates = [VirtualMachine.get(api.api, name=x)
-                         for x in api.list_template() if "miq-nightly" in x]
-    nightly_templates.sort(key=lambda x: datetime.datetime.strptime(x.name[-12:], "%Y%m%d%H%M"))
+    nightly_templates = filter(lambda t: 'miq-nightly' in t, api.list_template())
+    nightly_templates.sort(key=lambda x: datetime.datetime.strptime(x[-12:], '%Y%m%d%H%M'))
 
     if len(nightly_templates) > MRU_NIGHTLIES:
         for template in nightly_templates[:-MRU_NIGHTLIES]:
```

Comment:
based on
```
>>> a = [1,2,3]
>>> a[:-1]
[1, 2]
>>> a[:-4]
[]
>>> 
```

i propose `templates_to_delete = nightly_templates[:-MRU_NIGHTLIES]` - neither the if not the loop are needed